### PR TITLE
Worked on calculate_outcomes.py

### PR DESCRIPTION
**Pull request by anushka255**
Enclosed calculate_outcomes.py in main function, modified little bit of codes, and created test case for it

### CODE REVIEW

File: NPS_generation/calculate_outcomes.py [hunk 1]
```diff
@@ -99,383 +60,425 @@ def pct_stereocentres(mol):
...
+                kl_rings1 = discrete_KL(gen_rings1, org_rings1)
+                kl_rings2 = discrete_KL(gen_rings2, org_rings2)
+                kl_rings3 = discrete_KL(gen_rings3, org_rings3)
+                jsd_rings1 = discrete_JSD(gen_rings1, org_rings1)
+                jsd_rings2 = discrete_JSD(gen_rings2, org_rings2)
+                jsd_rings3 = discrete_JSD(gen_rings3, org_rings3)
+                emd_rings1 = discrete_EMD(gen_rings1, org_rings1)
+                emd_rings2 = discrete_EMD(gen_rings2, org_rings2)
+                emd_rings3 = discrete_EMD(gen_rings3, org_rings3)
+                res = res._append(pd.DataFrame({
```
Comment (vineetbansal) [inline NPS_generation/calculate_outcomes.py:333]: We should use `res = res.concat([res, pd.DataFrame...], axis=1)

File: NPS_generation/functions.py [hunk 2]
```diff
@@ -30,7 +30,7 @@ def clean_mol(smiles, stereochem=False, selfies=False, deepsmiles=False):
             smiles = converter.decode(deepsmiles)
         except:
             raise ValueError("invalid DeepSMILES: " + str(deepsmiles))
-    mol = Chem.MolFromSmiles(str(smiles))
+    mol = Chem.MolFromSmiles(smiles)
```
Comment (vineetbansal) [inline NPS_generation/functions.py:33]: Let's try to keep this `str` since `smiles` is a numpy array.

File: tests/test_data/output_step1.smi [hunk 3]
```diff
@@ -1,1976 +1,1976 @@
-C=CC1(C)CCc2c3c(cc(O)c2C1O)C(C)(C)C(O)CC3
-CCc1cc(S(=O)(=O)NCCCCO)ccc1-c1ccc(F)cc1F
-O=C(c1ccc(OCc2ccc(CN3CCCCC3)cc2)cc1)C1CC1
-O=C(O)CN1CCCC(=C2N(CC(=O)O)c3ccccc3N2C2CC3CCCC(C2)N3C2CC3CCCCC(C3)C2)C1
-CCN(CC)C(=O)c1c(NC(=O)c2cccs2)sc2c1CCC2
-Cc1cn(-c2ccn3c(-c4cccc(NC(=O)NCC(F)(F)F)c4)cnc3c2)cn1
-Cn1nccc1-c1nc(Nc2n[nH]c3ccccc23)c2c(n1)C(C)(C)OC2
 O=C(Cc1c[nH]c2ccccc12)NN=Cc1ccco1
```
Comment (vineetbansal) [inline tests/test_data/output_step1.smi:1]: Let's see if we can use the file from before and still have everything pass.

File: tests/test_data/output_step2.smi [hunk 4]
```diff
@@ -1,1976 +1,1976 @@
...
-C(CC(C)C)(NP(=O)(Oc1ccccc1)OC)C(N)=O
-C1C(S(c2ccccc2)(=O)=O)CCN(Cc2ccccc2)C1
-CCCCCOc1ccc2c(c1)c(CCN)c[nH]2
-O=S(=O)(c1c(C)cc(C)cc1C)NCc1ccc(C(NCCN2Cc3ccccc3C2)=O)cc1
-c1(C)nc2n(cccc2)c1C(OCC)=O
-N(CCc1ccccc1)(C1CCN(CCC)CC1)Cc1ccccc1
-N(C(Cc1ccccc1)C(OC)=O)C(=O)CCc1[nH]c(=O)c2c(n1)cccc2
-N12CCC3(CC1c1c(cccc1)CC2)OC(=O)NC3
-c1(N)cc(O)nc(SCCCCC)n1
+C(c1c2ccccc2[nH]c1)C(NN=Cc1occc1)=O
```
Comment (vineetbansal) [inline tests/test_data/output_step2.smi:1]: Let's not change this file unless we have to.

**Comment by vineetbansal** [comment]
Let's rename `sample-1-SMILES.smi` to `input_step4.smi`.

**Comment by vineetbansal** [comment]
Let's get rid of `sample-1-SMILES-outcomes.csv.gz` since we're not using it for anything just yet.

File: tests/test_nps_generation.py [hunk 5]
```diff
@@ -61,3 +61,14 @@ def test_train_model():
...
     ]
     train_model_main(args_list)
+
+
+def test_calculate_outcomes():
+    sampled_file = os.path.join(test_dir, "sample-1-SMILES.smi")
+
+    args_list = [
+        '--original_file', original_file,
+        '--output_dir', test_dir,
```
Comment (vineetbansal) [inline tests/test_nps_generation.py:71]: We should use `temp_dir` as the output directory just like the other tests.

Comment (vineetbansal) [reply tests/test_nps_generation.py:71]: Same comment for `--output_dir` in `test_train_model` as wel..

File: NPS_generation/calculate_outcomes.py (hunk 1 above)
Comment (anushka255) [reply NPS_generation/calculate_outcomes.py]: Changed every occurrence of this. The correct code is `res = pd.concat([res, pd.DataFrame...], axis=1)`

File: NPS_generation/functions.py (hunk 2 above)
Comment (anushka255) [reply NPS_generation/functions.py]: Changed it.

File: tests/test_data/output_step1.smi (hunk 3 above)
Comment (anushka255) [reply tests/test_data/output_step1.smi]: Reverted it back to the original one.

File: tests/test_data/output_step2.smi (hunk 4 above)
Comment (anushka255) [reply tests/test_data/output_step2.smi]: Reverted it back to the original one.

File: tests/test_nps_generation.py (hunk 5 above)
Comment (anushka255) [reply tests/test_nps_generation.py]: Changed both cases to account for temp_dir.

File: NPS_generation/calculate_outcomes.py (hunk 1 above)
Comment (vineetbansal) [reply NPS_generation/calculate_outcomes.py]: Looks like it should be `axis=0` instead. Sorry about that. Let me fix and push.

**Pull request merged**